embeddings/src/lib.rs

```diff
@@ -38,97 +37,203 @@ fn handle_embeddings(req: Request) -> Result<Response> {
     router.handle(req)
 }
 
-fn get_embeddings(_req: Request, _params: Params) -> Result<Response> {
-    // Return all embeddings from SQL
+fn get_embeddings(req: Request, _params: Params) -> Result<Response> {
+    match req.uri().query().is_some() {
+        true => {
+            let query = serde_qs::from_str(req.uri().query().unwrap_or_default()).unwrap();
 
-    let query = "SELECT * FROM embeddings";
-    let conn = Connection::open_default()?;
-    let result = conn.execute(query, &[])?;
-    info!("Result: {:?}", result);
-    let embedding_records: Vec<Embedding> = conn
-        .execute(query, &[])?
-        .rows()
-        .map(|row| -> anyhow::Result<Embedding> { row.try_into() })
-        .collect::<anyhow::Result<Vec<Embedding>>>()?;
-
-    info!("Rows: {:?}", embedding_records);
+            let result_set = get_similar(query);
 
-    Ok(http::Response::builder()
-        .status(http::StatusCode::OK)
-        .body(Some(serde_json::to_string(&embedding_records)?.into()))
-        .unwrap())
+            Ok(http::Response::builder()
+                .status(http::StatusCode::OK)
+                .header("Content-Type", "application/json")
+                .body(Some(serde_json::to_vec(&result_set.unwrap())?.into()))
+                .unwrap())
+        }
+        false => {
+            info!("In else");
+            let query = "SELECT * FROM embeddings";
+            let conn = Connection::open_default()?;
+            let embedding_records: Vec<Embedding> = conn
+                .execute(query, &[])?
+                .rows()
+                .map(|row| -> anyhow::Result<Embedding> { row.try_into() })
+                .collect::<anyhow::Result<Vec<Embedding>>>()?;
+
+            trace!("Rows: {:?}", embedding_records);
+            Ok(http::Response::builder()
+                .status(http::StatusCode::OK)
+                .body(Some(serde_json::to_string(&embedding_records)?.into()))
+                .unwrap())
+        }
+    }
 }
 
-fn create_embeddings(_req: Request, _params: Params) -> Result<Response> {
-    // Serialize the body of the req to Vec<embedding>
-    // Create a SQL query to search for existing and redact
-    // Submit the delta to SQL
+fn create_embeddings(req: Request, _params: Params) -> Result<Response> {
+    let embedding_request: EmbeddingRequest = serde_json::from_slice(
+        req.body()
+            .as_ref()
+            .map(|b| -> &[u8] { b })
```

Comment:
This might work instead:
```suggestion
            .as_deref()
```